xbmc/video/ContextMenus.cpp

```diff
@@ -119,7 +120,17 @@ static void SetPathAndPlay(CFileItem& item)
     item.SetPath(item.GetVideoInfoTag()->m_strFileNameAndPath);
   }
   item.SetProperty("check_resume", false);
-  g_application.PlayMedia(item, "", PLAYLIST_NONE);
+
+  CFileItemPtr movieItem((std::make_shared<CFileItem>(item)));
```

Comment:
It is more efficient because `std::make_shared` only requires a single heap allocation whereas `CFileItemPtr(new CFileItem())` requires two heap allocations (one for the reference counter and one for the item itself).

And yes it can throw exceptions itself but in that case the `std::shared_ptr` instance hasn't been created yet so there's no memory leak. If you however do something like
```
SomeMethod(CFileItemPtr(new CFileItem("foo")), CFileItemPtr(new CFileItem("bar")))
```
the compiler can decide in what order to execute the statements. He could choose to do
1. `new CFileItem("foo")`
2. `new CFileItem("bar")`
3. `CFileItemPtr(foo)`
4. `CFileItemPtr(bar)`

and for some reason you get an exception in step 2 there will be a memory leak because step 1 has already been executed and you can't get to that memory anymore. This can't happen with
```
SomeMethod(std::make_shared<CFileItem>("foo"), std::make_shared<CFileItem>("bar"))
```